contracts/VoteLockerCurve.sol

```diff
@@ -103,38 +100,19 @@ contract VoteLockerCurve is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     ///@notice userEpoch important for fetching previous block per user voting power
     mapping(address => uint256) public userEpoch;
 
-    /* @notice Global Checkpoints part of the equation to define combined voting
-     * power of all users.
-     */
-    Checkpoint[] private _globalCheckpoints;
-    ///@notice globalEpoch important for fetching previous block combined voting power
-    uint256 public globalEpoch;
-
     ///@notice Lockup mapping for each user
     mapping(address => Lockup) public lockups;
 
-    /* @notice slopeChanges part of the equation to define combined voting power of
-     * all users. Slope changes always complement only the latest global Checkpoint and are
-     * not used when fetching combined voting power of previous blocks.
-     */    
-    mapping(uint256 => int128) public slopeChanges;
+    ///@notice Contains group vote information used for totalSupply & totalSupplyAt
+    GroupVotePowerState public totalVotes;
```

Comment:
What is the approximate gas cost for a user lockup action now that we have to update both the `totalVotes` and `userTotalVotes` structures?

Reply:
good point thanks for bringing this up, have updated that info in the original PR description